ports/geckolib/gecko_bindings/sugar/ns_style_coord.rs

```diff
@@ -0,0 +1,29 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use structs::{nsStyleCoord_CalcValue, nsStyleCoord_Calc, nsStyleUnit, nsStyleUnion};
+use bindings::{Gecko_ResetStyleCoord, Gecko_SetStyleCoordCalcValue};
+
+// Functions here are unsafe because it is possible to use the wrong nsStyleUnit
```

Comment:
I'd like to understand the difference between `unsafe` on the function interface vs wrapping the body of the function in `unsafe` here.  Is it because we have a requirement that is not expressible in the type system, which must be adhered to otherwise we will have a memory safety issue, namely that you must pass in a reference to the `nsStyleUnit` associated with the `nsStyleUnion` and not just any one you have access to?  And to signal to users of `nsStyleUnion` that we have other requirements the compiler won't catch, we use `unsafe` here?